htmlreport/info_from_cmd.py

```diff
@@ -0,0 +1,59 @@
+import subprocess
+import os
+import datetime
+import locale
+
+
+def get_results(line, path, file, blame_options):
+    head, tail = os.path.split(file)
+    if head != "":
+        path = head
+    try:
+        os.chdir(path)
+    except:
+        return ""
+    try:
+        result = subprocess.check_output(f'git blame -L {str(line)}{" -w" if "-w" in blame_options else ""}{" -M" if "-M" in blame_options else ""} --porcelain -- {file}')
+        result = result.decode(locale.getpreferredencoding())
+    except:
+        return ""
+
+    return result
+
+
+def get_author(text):
+    author = text[text.find('author') + 7:text.find('author-mail')-1]
+    if text.startswith('fatal') or author == '':        # if error occurs from cmd starting with fatal sets the author to 'unknown'
+        author = 'Unknown'
+    disallowed_characters = '\<>'
+
+    for character in disallowed_characters:
+        author = author.replace(character, "")
+
+    return author
+
+
+def get_author_mail(text):
+    author_mail = text[text.find('author-mail') + 13:text.find('author-time')-2]
```

Comment:
this text processing looks like a hack. imagine that author-mail does not exist and author-time is located at position 20.. then author_mail will be some junk..

how about splitting git blame output into a dict in the `get_results`.

    git_blame_output='author-mail=fred.flintstone author-time=12345'

=>

    d['author-mail'] = 'fred.flintstone'
    d['author-time'] = '12345'

Reply:
I have created a dict from git_blame. 
Git blame returns a massive string and it is very hard to parse that data. That's why I used text.find() and tried to get around parsing that string... I have changed parsing that data a bit, right now it should not be a problem even if the author mail is unknown.